mruby/src/value/mod.rs

```diff
@@ -75,6 +80,32 @@ impl Value {
     }
 }
 
+impl Clone for Value {
+    fn clone(&self) -> Self {
+        let ruby_type = self.ruby_type();
+        if ruby_type == Data {
+            unsafe {
+                let ptr = mem::transmute::<*mut c_void, Rc<RefCell<_>>>(self.value.value.p);
+                let cloned_ptr = Rc::clone(&ptr);
+                Self::new(
+                    Rc::clone(&self.interp),
+                    sys::mrb_value {
```

Comment:
this is mostly right, but I feel more comfortable using mruby macros to muck around with the union. We can get an uninitialized `mrb_value` that is an `MRB_TT_DATA` by calling `mrb_sys_data_value` and then initialize it with our `void *` using `mrb_sys_data_init`:

https://github.com/lopopolo/ferrocarril/blob/37034d455d75d39eae4be5ffdd458de98713e42b/mruby-sys/mruby-sys/src/mruby-sys/ext.c#L93-L100

https://github.com/lopopolo/ferrocarril/blob/37034d455d75d39eae4be5ffdd458de98713e42b/mruby-sys/mruby-sys/src/mruby-sys/ext.c#L133-L135

Reply:
It looks like `mrb_sys_data_init` is not the right C extension to use since we don't know the `data_type`. We should expose a data set via `DATA_PTR` macro.

https://github.com/lopopolo/ferrocarril/blob/37034d455d75d39eae4be5ffdd458de98713e42b/mruby-sys/vendor/mruby-c078758/include/mruby/data.h#L66-L72